src/index.js

```diff
@@ -36,10 +47,19 @@ class LndEngine {
     this.macaroonPath = macaroonPath
     this.protoPath = LND_PROTO_FILE_PATH
 
-    this.client = generateLndClient(this.host, this.protoPath, this.tlsCertPath, this.macaroonPath)
+    this.client = generateLightningClient(this)
+    this.walletUnlocker = generateWalletUnlockerClient(this)
 
     Object.assign(this, actions)
   }
+
+  /**
+   * Client will return false if the lnd-engine requires wallet configuration before
+   * the engine is fully functional.
+   */
+  get isUnlocked () {
+    return !!Object.keys(this.client).length
```

Comment:
This seems like a pretty strange way to detect the unlocked status.

A more understandable way to do this (if it's possible) would be to do something like try the client and update a specific property (e.g. `isUnlocked`) if it responds with a particular error code.

Reply:
I wish it gave us a specific error code. We'll have to think about this. What happens is all of our `client` methods become undefined. GRPC will return a blank object for `Lightning` which is why we check for the length of keys (which would be 0 and false)

Reply:
@treygriffith this has been fixed in recent commits. Thank you for pushing on a more thorough investigation